qutebrowser/browser/commands.py

```diff
@@ -303,6 +303,14 @@ def openurl(self, url=None, related=False,
                 tab = False
                 bg = True
 
+            if utils.url_in_tabs(cur_url):
+                index = utils.tuple_for_url_in_tab(cur_url.toDisplayString())[2]
```

Comment:
This snippet is looping over all tabs 4 times

- generating the urls in `list_buffers`
- searching generated list in `url_in_tabs`
- generating the urls again in `list_buffers`
- equality check in `tuple_for_url_in_tab`.

If there's no way around avoiding the iteration completely, please just return the tab in the first loop, or if you must, generate a map from url->tab.

Reply:
This is now iterating only once.